### ISSUE TITLE
`--arch` unhelpful on macOS because `get_safe_arch` resets it

### ISSUE THREAD

**Issue by Zegnat** [Schniz/fnm#592]
I was able to make older Node.js builds available for ARM (Apple Silicon / M1) on an internal mirror (provided to fnm with `--node-dist-mirror`) but it is impossible to get it to download.

I expected the following command to work, but as you can see by the output, it does not:

```bash
fnm install --arch arm64 --node-dist-mirror https://example.com/ 12
# Installing Node v12.22.7 (x64)
# error: Can't download the requested binary: v12.22.7 for x64 not found upstream.
# You can `fnm ls-remote` to see available versions or try a different `--arch`.
```

It took me a while to figure out that fnm was blocking this, especially because the error message kept telling me to try a different `--arch`. Turns out you are unable to use a different `--arch` because `get_safe_arch` will *always* overwrite my choice:

https://github.com/Schniz/fnm/blob/cc158ea44e9df45f0d9961dd482296e5181eafc1/src/arch.rs#L19-L22

It seemed counter intuitive to me that a manually supplied arch is being overwritten. Surely the worst that can happen is the user providing an arch that Node.js does not offer in their releases and the installation immediately failing? I understand that the arch should default to x64 when someone does a pure `fnm install 12`, but IMO it should not change the arch if it was manually provided.

Was there any reason for this? Would a PR that does not run `get_safe_arch` on top of user provided `--arch` be accepted?

**Comment by Schniz** [comment]
That's a good question. I think we provide `FNM_ARCH` with `fnm env`, so the users pass `--arch` implicitly with environment variables. Maybe we need to set `FNM_FORCE_ARCH`? Maybe we need to avoid the special casing and try to download arm64 and fallback to x64 if the file is missing? https://nodejs.org/dist/index.json has `files` that we don't use right now.

**Comment by Schniz** [comment]
So anyway I think that the change will not be trivial, but we might want to alter the behavior to look for the files that the node version index provides.

**Comment by Zegnat** [comment]
I agree that the true answer is probably in the `files` array. That change is sadly outside of what I feel I could comfortably whip up in Rust.

Would you accept a PR for an `FNM_FORCE_ARCH` first? If so: would you expect only the environment variable, a `--force-arch` flag option, or both?

**Comment by trm217** [comment]
When running `fnm install` with the `--arch` flag, the behaviour seems weird all together.
I was trying to install Node 14 for arm64 and it just heads straight for installing Node 14 for x64 without any info on that the targeted installation was not available.

**Comment by natew** [comment]
Ran into this today, it seems 14 does have arm builds, is there a reason it doesn't allow installing it? The automatic regression to x64 is very unintuitive.

**Comment by mshick** [comment]
Just encountered this as a fairly new user of `fnm`. Took me awhile to figure out why `--arch` was being ignored for v14. Was loving `fnm` but not being able to install an `arm64` node 14 is a blocker. 

Big +1 for clarifying this limitation, and perhaps offering a way to force the arch as suggested. Hoping this issue doesn't fall off the radar.

**Comment by Schniz** [comment]
I had a thought while driving: what if we had `fnm import-installation ./my-self-compiled-node-directory`?

**Comment by y-nk** [comment]
we could also allow trust or 3rd party proposing builds such as https://github.com/a60814billy/node14-apple-silicon-build

**Comment by Schniz** [comment]
@y-nk  good idea. You can already set the node dist mirror but we don’t support index.tab — we can support the tsv file and then it’ll work.
